### DISCUSSION TITLE
Deployment Failed for https://github.com/alshedivat/al-folio.git

### DISCUSSION THREAD

**Discussion by Rai-Sahil** [community/community#123020] [Actions]
### Select Topic Area

Question

### Body

Logs:
```
Warning: the running version of Bundler (2.1.4) is older than the version that created the lockfile (2.5.7). We suggest you to upgrade to the version that created the lockfile by running `gem install bundler:2.5.7`.
Bundler can't satisfy your Gemfile's dependencies.
Install missing gems with `bundle install`.
Warning: The github-pages gem can't satisfy your Gemfile's dependencies. If you want to use a different Jekyll version or need additional dependencies, consider building Jekyll site with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-actions/
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
  Liquid Exception: Liquid syntax error (line 60): Unknown tag 'toc' in /_layouts/post.liquid
/usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block.rb:38:in `unknown_tag': Liquid syntax error (line 60): Unknown tag 'toc' (Liquid::SyntaxError)
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:38:in `unknown_tag'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block.rb:68:in `block in parse_body'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:32:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block.rb:58:in `parse_body'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:30:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tag.rb:10:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:34:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/document.rb:10:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/document.rb:5:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:130:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:114:in `parse'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/liquid_renderer/file.rb:13:in `block in parse'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/liquid_renderer/file.rb:49:in `measure_time'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/liquid_renderer/file.rb:12:in `parse'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/renderer.rb:121:in `render_liquid'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/renderer.rb:195:in `render_layout'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/renderer.rb:158:in `place_in_layouts'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/renderer.rb:88:in `render_document'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/renderer.rb:62:in `run'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/site.rb:479:in `render_regenerated'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/site.rb:464:in `block (2 levels) in render_docs'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/site.rb:463:in `each'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/site.rb:463:in `block in render_docs'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/site.rb:462:in `each_value'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/site.rb:462:in `render_docs'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/site.rb:191:in `render'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/site.rb:71:in `process'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/command.rb:28:in `process_site'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/commands/build.rb:65:in `build'
	from /usr/local/bundle/gems/jekyll-3.9.5/lib/jekyll/commands/build.rb:36:in `process'
	from /usr/local/bundle/gems/github-pages-231/bin/github-pages:70:in `block (3 levels) in <top (required)>'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	from /usr/local/bundle/gems/github-pages-231/bin/github-pages:6:in `<top (required)>'
	from /usr/local/bundle/bin/github-pages:23:in `load'
	from /usr/local/bundle/bin/github-pages:23:in `<main>'
Error:  Logging at level: debug Configuration file: /github/workspace/./_config.yml GitHub Pages: github-pages v231 GitHub Pages: jekyll v3.9.5 Theme: jekyll-theme-primer Theme source: /usr/local/bundle/gems/jekyll-theme-primer-0.6.0 Requiring: jekyll-github-metadata Requiring: jekyll-seo-tag Requiring: jekyll-feed Requiring: jekyll-sitemap Requiring: jemoji Requiring: jekyll-coffeescript Requiring: jekyll-commonmark-ghpages Requiring: jekyll-gist Requiring: jekyll-github-metadata Requiring: jekyll-paginate Requiring: jekyll-relative-links Requiring: jekyll-optional-front-matter Requiring: jekyll-readme-index Requiring: jekyll-default-layout Requiring: jekyll-titles-from-headings GitHub Metadata: Initializing... Source: /github/workspace/. Destination: /github/workspace/./_site Incremental build: disabled. Enable with --incremental Generating... EntryFilter: excluded /LICENSE EntryFilter: excluded /FAQ.md EntryFilter: excluded /README.md EntryFilter: excluded /docker-compose-slim.yml EntryFilter: excluded /package.json EntryFilter: excluded /package-lock.json EntryFilter: excluded /docker-compose.yml EntryFilter: excluded /purgecss.config.js EntryFilter: excluded /INSTALL.md EntryFilter: excluded /CUSTOMIZE.md EntryFilter: excluded /Dockerfile EntryFilter: excluded /Gemfile.lock EntryFilter: excluded /Gemfile EntryFilter: excluded /CONTRIBUTING.md Reading: _posts/2024-01-27-code-diff.md Reading: _posts/2021-07-04-diagrams.md Reading: _posts/2015-10-20-math.md Reading: _posts/2023-04-25-sidebar-table-of-contents.md Reading: _posts/2024-04-15-pseudocode.md Reading: _posts/2020-09-28-github-metadata.md Reading: _posts/2022-12-10-giscus-comments.md Reading: _posts/2023-03-21-tables.md Reading: _posts/2024-01-26-echarts.md Reading: _posts/2023-03-20-table-of-contents.md Reading: _posts/2023-12-12-tikzjax.md Reading: _posts/2024-04-29-typograms.md Reading: _posts/2024-01-27-advanced-images.md Reading: _posts/2023-07-04-jupyter-notebook.md Reading: _posts/2015-05-15-images.md Reading: _posts/2018-12-22-distill.md Reading: _posts/2022-02-01-redirect.md Reading: _posts/2015-07-15-code.md Reading: _posts/2024-04-28-post-citation.md Reading: _posts/2023-04-25-audios.md Reading: _posts/2020-09-28-twitter.md Reading: _posts/2023-07-12-post-bibliography.md Reading: _posts/2023-04-24-videos.md Reading: _posts/2024-01-26-chartjs.md Reading: _posts/2015-03-15-formatting-and-links.md Reading: _posts/2023-05-12-custom-blockquotes.md Reading: _posts/2024-01-27-vega-lite.md Reading: _posts/2024-05-01-tabs.md Reading: _posts/2024-01-26-geojson-map.md Reading: _posts/2015-10-20-disqus-comments.md EntryFilter: excluded /lighthouse_results/desktop EntryFilter: excluded /lighthouse_results/mobile EntryFilter: excluded /bin/entry_point.sh EntryFilter: excluded /bin/deploy EntryFilter: excluded /bin/cibuild EntryFilter: excluded /_pages/about_einstein.md EntryFilter: excluded /readme_preview/photos-screenshot.png EntryFilter: excluded /readme_preview/dark.png EntryFilter: excluded /readme_preview/blog.png EntryFilter: excluded /readme_preview/distill.png EntryFilter: excluded /readme_preview/people.png EntryFilter: excluded /readme_preview/cv.png EntryFilter: excluded /readme_preview/code.png EntryFilter: excluded /readme_preview/light.png EntryFilter: excluded /readme_preview/math.png EntryFilter: excluded /readme_preview/repositories.png EntryFilter: excluded /readme_preview/projects.png EntryFilter: excluded /readme_preview/publications.png EntryFilter: excluded /readme_preview/jupyter.png EntryFilter: excluded /readme_preview/al-folio-preview.png Reading: _news/announcement_3.md Reading: _news/announcement_2.md Reading: _news/announcement_1.md Reading: _projects/6_project.md Reading: _projects/3_project.md Reading: _projects/4_project.md Reading: _projects/1_project.md Reading: _projects/5_project.md Reading: _projects/2_project.md Generating: JekyllOptionalFrontMatter::Generator finished in 5.5394e-05 seconds. Generating: JekyllReadmeIndex::Generator finished in 5.9071e-05 seconds. Jekyll Feed: Generating feed for posts Generating: JekyllFeed::Genera
```

Is there any fixes for this? I think it's bundle version error, but when I tried to update the version that fails too.

**Comment by Jaaziel-Polanco** [comment]
It looks like you’re encountering several issues with your Jekyll site deployment on GitHub Pages. Here’s a step-by-step guide to address each of the errors:

Bundler Version Warning:
The warning indicates that the version of Bundler you’re running is older than the one that created the Gemfile.lock. To resolve this, update Bundler to the specified version:
gem install bundler:2.5.7

Gemfile Dependencies:
If Bundler can’t satisfy your Gemfile’s dependencies, run bundle install to install any missing gems. If this fails, you may need to update your Gemfile.lock by deleting it and running bundle install again.
GitHub Pages Dependency Issues:
The GitHub Pages gem may not be compatible with your current Gemfile’s dependencies. [Consider using GitHub Actions for building your Jekyll site, which allows more flexibility with dependencies](https://stackoverflow.com/questions/75912209/build-error-while-trying-to-publish-my-github-pages)[1](https://stackoverflow.com/questions/75912209/build-error-while-trying-to-publish-my-github-pages).
Faraday Retry Middleware:
For Faraday v2.0+, you need to install the faraday-retry gem to use retry middleware. This can help with intermittent network issues:
gem install faraday-retry

Liquid Syntax Error:
The ‘toc’ tag error suggests that the jekyll-toc plugin might not be supported by GitHub Pages or is not included in your Gemfile. [If you’re using a custom plugin for generating a table of contents, ensure it’s supported by GitHub Pages or consider using a different method](https://stackoverflow.com/questions/75912209/build-error-while-trying-to-publish-my-github-pages)[2](https://github.com/alshedivat/al-folio/issues/1398).
General Troubleshooting:
Ensure that your Ruby version matches the one required by GitHub Pages.
Check that all plugins used in your _config.yml are supported by GitHub Pages.
If you’re using a theme, ensure it’s compatible with GitHub Pages.
Remember to commit and push any changes you make to your repository to trigger a new build on GitHub Pages. If you continue to experience issues, you might want to build your site locally with bundle exec jekyll serve to debug any problems before pushing to GitHub. If you need further assistance, please provide more details about your setup and the steps you’ve taken so far.

**Comment by sundongwei** [comment]
I have the same problem. I do not know how to slove it ~

**Comment by formozov** [comment]
I had a similar problem. Solved by following instructions in the file (Enabling automatic deployment and not Recommended Approach): https://github.com/formozov/formozov.github.io/blob/master/INSTALL.md.

**Comment by github-actions** [comment]
🕒 **Discussion Activity Reminder** 🕒

This Discussion has been labeled as dormant by an automated system for having no activity in the last 60 days. Please consider one the following actions:

1️⃣ Close as Out of Date: If the topic is no longer relevant, close the Discussion as `out of date` at the bottom of the page.

2️⃣ Provide More Information: Share additional details or context — or let the community know if you've found a solution on your own.

3️⃣ Mark a Reply as Answer: If your question has been answered by a reply, mark the most helpful reply as the solution.

Note: This dormant notification will only apply to Discussions with the `Question` label. To learn more, see our [recent announcement](https://github.com/orgs/community/discussions/70478).

Thank you for helping bring this Discussion to a resolution! 💬
